frontend/src/features/admin-form/create/common/CreatePageSidebarContext/CreatePageSidebarContext.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  createContext,
+  FC,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from 'react'
+
+import {
+  setToInactiveSelector,
+  useBuilderAndDesignStore,
+} from '../../builder-and-design/useBuilderAndDesignStore'
+
+export enum DrawerTabs {
+  Builder,
+  Design,
+  Logic,
+}
+
+type CreatePageSidebarContextProps = {
+  activeTab: DrawerTabs | null
+  isDrawerOpen: boolean
+  handleBuilderClick: () => void
+  handleDesignClick: () => void
+  handleLogicClick: () => void
+  handleClose: () => void
+}
+
+const CreatePageSidebarContext = createContext<
+  CreatePageSidebarContextProps | undefined
+>(undefined)
+
+export const useCreatePageSidebar = (): CreatePageSidebarContextProps => {
+  const context = useContext(CreatePageSidebarContext)
+  if (!context) {
+    throw new Error(
+      `useCreatePageSidebar must be used within a CreatePageSidebarProvider component`,
+    )
+  }
+  return context
+}
+
+export const useCreatePageSidebarContext =
+  (): CreatePageSidebarContextProps => {
+    const [activeTab, setActiveTab] = useState<DrawerTabs | null>(null)
+    const isDrawerOpen = useMemo(
+      () => activeTab !== null && activeTab !== DrawerTabs.Logic,
+      [activeTab],
+    )
+    const setFieldsToInactive = useBuilderAndDesignStore(setToInactiveSelector)
+
+    const handleBuilderClick = useCallback(
+      () => setActiveTab(DrawerTabs.Builder),
+      [setActiveTab],
+    )
+
+    const handleDesignClick = useCallback(() => {
+      setActiveTab(DrawerTabs.Design)
+      setFieldsToInactive()
```

Comment:
Rereading this; maybe we should put these invocations in a `useEffect` whenever active tab changes instead.

```tsx
useEffect(() => {
  setFieldsToInactive()
}, [activeTab])
```

Reply:
8e8eadb4